iib/workers/tasks/build.py

```diff
@@ -792,7 +799,83 @@ def _copy_files_from_image(image, src_path, dest_path):
             log.exception(e)
 
 
-def _adjust_operator_manifests(manifests_path):
+def _add_package_name_suffix(metadata_path, organization=None):
+    """
+    Add the package name suffix if configured for this organization.
+
+    This adds the suffix to the value of
+    ``annotations['operators.operatorframework.io.bundle.package.v1']`` in
+    ``metadata/annotations.yaml``.
+
+    The final package name value is returned as part of the tuple.
+
+    :param str metadata_path: the path to the bundle's metadata directory.
+    :param str organization: the organization this customization is for.
+    :raise IIBError: if the ``metadata/annotations.yaml`` file is in an unexpected format.
+    :return: a tuple with the package name and a dictionary of labels to set on the bundle.
+    :rtype: tuple(str, dict)
+    """
+    annotations_yaml_path = os.path.join(metadata_path, 'annotations.yaml')
+    if not os.path.exists(annotations_yaml_path):
+        raise IIBError('metadata/annotations.yaml does not exist in the bundle')
+
+    with open(annotations_yaml_path, 'r') as f:
+        try:
+            annotations_yaml = yaml.safe_load(f)
+        except yaml.YAMLError:
+            error = 'metadata/annotations/yaml is not valid YAML'
+            log.exception(error)
+            raise IIBError(error)
+
+    package_label = 'operators.operatorframework.io.bundle.package.v1'
+    if not isinstance(annotations_yaml.get('annotations', {}), dict):
```

Comment:
I find it difficult to follow what the code is trying to do when there's so much validation around it. Utility methods can help abstract some of the implementation details and help readers focus on what the code is trying to achieve.

```python
package_name = get_package_name_from_annotations(annotations_yaml)
package_name_with_suffix = apply_package_name_suffix(package_name)
```

There's nothing wrong with the approach you have taken. The code works and it's maintainable. This is just a suggestion to make it more readable which always has a subjective aspect to it.

Reply:
I think it's fine the way it is. If you feel strongly about this, we can discuss this further.